### PR TITLE
Update deployment and add version notice

### PR DESCRIPTION
**Pull request by Shotgunosine**
Addresses #76. `docker pull bids/freesurfer`, `docker pull bids/freesurfer:v6`, and `docker pull bids/freesurfer:v6-${CIRCLETAG}` will pull FreeSurfer 6.     
`docker pull bids/freesurfer:v7`, and `docker pull bids/freesurfer:v7-${CIRCLETAG}` will pull FreeSurfer 7.

I've added a warning as well. If you're ok with this arrangment @Remi-Gau, I can also update the readme in this PR.

### CODE REVIEW

**Comment by Remi-Gau** [comment]
on second thought not sure that this naming won't be confusing for users

right now this would give something like:

```bash
docker pull bids/freesurfer:v6-7.4.1
```

Should we try something "simpler"?

```bash
docker pull bids/freesurfer:${FSversion}
```

with FSVersion that could be one of:
- "6.0.1" 
- "7.4.1"

Or any version we decide to release

When we push to the default branch we release a

bids/freesurfer:${FSversion}_unstable for each FSversion

And for now latest matches "6.0.1" 

And in 2024 it will match "7.4.1"

**Comment by Remi-Gau** [comment]
If we want to make sure we keep all the previous versions (preferable IMHO) then the release tag should be the date and we would have:

- docker pull bids/freesurfer:7.4.1_202401
- docker pull bids/freesurfer:7.4.1_202309
- docker pull bids/freesurfer:7.4.1 would give you the latest release of freesurfer 7.4.1

- docker pull bids/freesurfer:6.0.1_202401
- docker pull bids/freesurfer:6.0.1_202309
- docker pull bids/freesurfer:6.0.1 would give you the latest release of freesurfer 6.0.1

- docker pull bids/freesurfer and - docker pull bids/freesurfer:latest would give you latest of 6.0.1 for now

**Comment by Shotgunosine** [comment]
Happy to have release tag as the date. I just don't know if we want to name the Freesurfer 6 and Freesurfer 7 versions with the whole version number. That implies that we're going to continue supporting multiple Freesurfer 7 versions at the same time. How about as a compromise we have the following:

- docker pull bids/freesurfer:7.4.1_202401

- docker pull bids/freesurfer:7.4.1_202309

- docker pull bids/freesurfer:7 would give you the latest release of Freesurfer 7

- docker pull bids/freesurfer:6.0.1_202401

- docker pull bids/freesurfer:6.0.1_202309

- docker pull bids/freesurfer:6 would give you the latest release of freesurfer 6

- docker pull bids/freesurfer and - docker pull bids/freesurfer:latest would give you latest of 6.0.1 for now

**Comment by Remi-Gau** [comment]
> That implies that we're going to continue supporting multiple Freesurfer 7 versions at the same time

good point
as long we have what FS version is in a given image I am totally OK making it clear we will NOT maintain all possible FS versions

File: .circleci/config.yml [hunk 1]
```diff
@@ -198,20 +198,31 @@ jobs:
 
               # make sure we have a lowercase repo
               user_name="bids"
-              repo_name=$(echo "${CIRCLE_PROJECT_REPONAME}_fs<< parameters.version >>" | tr '[:upper:]' '[:lower:]')
+              repo_name=$(echo "${CIRCLE_PROJECT_REPONAME}" | tr '[:upper:]' '[:lower:]')
+              FSMAJORVERSION=<< parameters.version >>
```
Comment (Remi-Gau) [inline .circleci/config.yml]: this is getting hard to read

should we go for SCREAMING_SNAKE_CASE?

FSVERSION -> FS_VERSION
FSMAJORVERSION -> FS_MAJOR_VERSION

Comment (Shotgunosine) [reply .circleci/config.yml]: Sure, fine by me.

**Comment by Remi-Gau** [comment]
OK good with me merging this one

**Comment by Shotgunosine** [comment]
Alright, if you give it a date based version tag it should deploy containers the way you wanted.

**Comment by bpinsard** [comment]
It seems that it didn't run the deploy step to dockerhub, I cannot see the tags. Is that expected?

**Comment by Remi-Gau** [comment]
yeah it is because I did not make a release, let me fix that

**Comment by bpinsard** [comment]
Cool thanks!

**Comment by bpinsard** [comment]
Looks like deploy is broken with the tags renaming.

**Comment by Shotgunosine** [comment]
@Remi-Gau, looks like an easy fix, just need to add:
`docker tag "${user_name}/${repo_name}" "${user_name}/${repo_name}:latest"`
before line 212 of config.yml

Do you want to just make that correction and push it directly to master?

**Comment by Remi-Gau** [comment]
will take care of it

**Comment by bpinsard** [comment]
gentle ping :D

**Comment by Remi-Gau** [comment]
> gentle ping :D

@bpinsard

new version released are on dockerhub: https://hub.docker.com/r/bids/freesurfer/tags

**Comment by bpinsard** [comment]
Thanks a lot! Will deploy it asap.

**Comment by SamGuay** [comment]
Amazing, thank you @Remi-Gau, I was patiently waiting for this release as well!

**Pull request merged**